app/javascript/angular/code/services/google/_map.js

```diff
@@ -149,15 +151,35 @@ export const map = (
       return newMarker;
     },
 
-    drawCustomMarker: function({ latLng, content, colorClass, callback, type }) {
-      const customMarker = buildCustomMarker(latLng, content, colorClass, callback, type);
+    drawCustomMarker: function({ latLng, content, colorClass, callback, type, objectId, value }) {
+      const customMarker = buildCustomMarker(latLng, content, colorClass, callback, type, objectId, value );
 
       customMarker.setMap(this.get());
       this.markers.push(customMarker);
 
       return customMarker;
     },
 
+    clusterMarkers: function(onClick) {
+      const options = {
+        styles: [
+          { url: '/assets/marker1.png', height: 10, width: 10 },
+          { url: '/assets/marker2.png', height: 10, width: 10 },
+          { url: '/assets/marker3.png', height: 10, width: 10 },
+          { url: '/assets/marker4.png', height: 10, width: 10 },
+        ],
+        zoomOnClick: false,
+        calculator: (markers, stylesCount) => {
+          const average = markers.reduce((sum, marker) => sum + marker.value(), 0) / markers.length
+          return { text: "", index: heat.getLevel(Math.round(average)) }
+        }
+      };
+
+      const markerClusterer = new MarkerClusterer(this.mapObj, this.markers, options);
+
+      googleMaps.listen(markerClusterer, 'clusterclick', onClick);
+    },
```

Comment:
I couldn't figure out how to test it, even using sinon. I can't find a way to stub MarkerClusterer. On the other hand, this method just basically creates some settings and passes it to the library, which I'm assuming is tested by google.